hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java

```diff
@@ -449,11 +389,21 @@ private void logAuditEvent(boolean succeeded, UserGroupInformation ugi,
   private final boolean erasureCodingEnabled;
   private final ErasureCodingManager erasureCodingManager;
 
-  private static int DB_ON_DISK_FILE_MAX_SIZE;
-  private static int DB_ON_DISK_SMALL_FILE_MAX_SIZE;
-  private static int DB_ON_DISK_MEDIUM_FILE_MAX_SIZE;
-  private static int DB_ON_DISK_LARGE_FILE_MAX_SIZE;
-  private static int DB_IN_MEMORY_FILE_MAX_SIZE;
+  static int DB_IN_MEMORY_BUCKET_SIZE;
+  static int DB_ON_DISK_SMALL_BUCKET_SIZE;
+  static int DB_ON_DISK_MEDIUM_BUCKET_SIZE;
+  static int DB_ON_DISK_LARGE_BUCKET_SIZE;
+  static int DB_MAX_SMALL_FILE_SIZE = 0;
+  static {
```

Comment:
Why do we need this static block? Won't we have that every time it is called the storage factory is not initialized?

Reply:
I don't understand the way GitHub highlights things, the block I am speaking about is the one that come after and is opened by the "static {" line 397

Reply:
> Why do we need this static block? Won't we have that every time it is called the storage factory is not initialized?

Well FSnamesystem can not be used without initializing the storage factory. The only time we use FSnamesystem without initializing the storage factory is in some unit tests what `mock` this class. 

Right row we do not have any good non-static way of getting parameter values from the database implementation layer.